traitsui/qt4/list_editor.py

```diff
@@ -221,10 +222,9 @@ def update_editor_item(self, event):
 
         # Otherwise, find the proxy for this index and update it with the
         # changed value:
-        for control in self.control.widget().children():
-            if isinstance(control, QtGui.QLayout):
-                continue
-
+        children = (self._layout.itemAt(ind).widget()
+                    for ind in range(self._layout.count()))
+        for control in children:
```

Comment:
A better way to do this would just be:
```
for control in self._list_pane.children():
```
I think.  Then you don't need to track the `_layout` specifically as a trait.